### ISSUE TITLE
Collaboration style?

### ISSUE THREAD

**Issue by ggjx22** [ai-practitioners/time-series-forecasting#2]
Let's set some guidelines to collaborate. @ellacharmed previously we tried to collaborate and I briefly recalled that you had trouble merging codes across 3 notebooks together?

Shall we all work on a single research.ipynb this time?

**Comment by ellacharmed** [comment]
We can work on our own branch, then compare and integrate whichever code is more _efficient_ or _elegant_ or _straightforward_ (whatever critieria), into the `main` branch, as and when?

Just means need to commit and push often.

**Comment by ggjx22** [comment]
sounds good. we compare and integrate during every mini milestone, then merge into `main`

**Comment by ellacharmed** [comment]
I've been rethinking this.

Maybe we should do it by atomic-tasks like in kanban board style instead of each of us doing a full EDA and then a full ML modeling coding, which then needs to be integrated?

So, we first need to list out each atomic task in kanban board, and then we take which task/feature we want to work on, assign it to ourselves, work on it and then commit that. A feature that can be done in a day or 2 days of say 2-4hrs, or the smallest feature that can live by itself aka - a "minimum viable product". 

However, I'm stumped as how to do this as none of the tasks can be in parallel mode, each task needs to be in sequential order...

**Comment by ggjx22** [comment]
we should have a better idea of how to split the task once we have the small steps layout. E.g. in terms of importing data phase, I can be the one writing the query in mysql workbench, you can be the one that writes the function/oop classes that handle the action of importing data. at the end of the day, we sit together again or discuss how to merge our tasks to complete the milestone - something like that?

in essence, we all are working on different features and towards the same milestone, never one phase behind or infront. i believe we can think about this once we have explored the data set sufficiently.
